src/actions/attack.cpp

```diff
@@ -122,11 +122,11 @@ battle_context_unit_stats::battle_context_unit_stats(const unit& u,
 	}
 
 	// Get the weapon characteristics as appropriate.
-	auto ctx = weapon->specials_context(u, opp, u_loc, opp_loc, attacking, opp_weapon);
+	auto ctx = weapon->specials_context(&u, &opp, u_loc, opp_loc, attacking, opp_weapon);
```

Comment:
I'd prefer if the address-of is done by the `specials_context_t` constructor instead of at the call site.

Reply:
I don't see the problem with adding another overload.